### ISSUE TITLE
Zoltan2_findUniqueGids_MPI_4 fails on haswell

### ISSUE THREAD

**Issue by jjellio** [trilinos/Trilinos#732]
This is an OpenMP build with LONG LONG and INT ordinal types. Built using the Cray environment for Intel (version 16.0.3 compilers).

This may be related to #730 

```
    Start 631: Zoltan2_findUniqueGids_MPI_4

631: Test command: /opt/cray/alps/6.1.6-20.1/bin/aprun "-e" "OMP_DISPLAY_ENV=true" "-e" "CRAY_OMP_CHECK_AFFINITY=true" "-e" "OMP_NUM_THREADS=16" "-e" "OMP_THREAD_PLACES=core" "-d" "8" "-j" "2" "-cc" "depth" "-n" "4" "/lscratch1/jjellio/trilinos-opt-shared-hsw/packages/zoltan2/test/unit/Zoltan2_findUniqueGids.exe"
631: Test timeout computed to be: 600
631: 
631: Teuchos::GlobalMPISession::GlobalMPISession(): started processor with name nid00015 and rank 1!
631: Teuchos::GlobalMPISession::GlobalMPISession(): started processor with name nid00015 and rank 0!
631: Teuchos::GlobalMPISession::GlobalMPISession(): started processor with name nid00015 and rank 3!
631: Teuchos::GlobalMPISession::GlobalMPISession(): started processor with name nid00015 and rank 2!
631: --------
631:   Starting  test1: int
631: [0] Zoltan ERROR in Zoltan_DD_Create (line 114 of /home/jjellio/src/Trilinos/packages/zoltan/src/Utilities/DDirectory/DD_Create.c):  Invalid input argument
631: [3] Zoltan ERROR in Zoltan_DD_Create (line 114 of /home/jjellio/src/Trilinos/packages/zoltan/src/Utilities/DDirectory/DD_Create.c):  Invalid input argument
631: [2] Zoltan ERROR in Zoltan_DD_Create (line 114 of /home/jjellio/src/Trilinos/packages/zoltan/src/Utilities/DDirectory/DD_Create.c):  Invalid input argument
631: [1] Zoltan ERROR in Zoltan_DD_Create (line 114 of /home/jjellio/src/Trilinos/packages/zoltan/src/Utilities/DDirectory/DD_Create.c):  Invalid input argument
631: [-1] Zoltan ERROR in Zoltan_DD_Update (line 93 of /home/jjellio/src/Trilinos/packages/zoltan/src/Utilities/DDirectory/DD_Update.c):  Invalid input argument
631: [-1] Zoltan ERROR in Zoltan_DD_Update (line 93 of /home/jjellio/src/Trilinos/packages/zoltan/src/Utilities/DDirectory/DD_Update.c):  Invalid input argument
631: [-1] Zoltan ERROR in Zoltan_DD_Update (line 93 of /home/jjellio/src/Trilinos/packages/zoltan/src/Utilities/DDirectory/DD_Update.c):  Invalid input argument
631: [-1] Zoltan ERROR in Zoltan_DD_Update (line 93 of /home/jjellio/src/Trilinos/packages/zoltan/src/Utilities/DDirectory/DD_Update.c):  Invalid input argument
631: _pmiu_daemon(SIGCHLD): [NID 00015] [c0-0c0s3n3] [Mon Oct 24 11:49:55 2016] PE RANK 2 exit signal Segmentation fault
631: Application 49008 exit codes: 139
631: Application 49008 resources: utime ~0s, stime ~1s, Rss ~20860, inblocks ~0, outblocks ~0
3/6 Test #631: Zoltan2_findUniqueGids_MPI_4 ...........***Failed    5.44 sec

```

```
rm -f CMakeCache.txt; rm -rf CMakeFiles


cmake \
   -D CMAKE_VERBOSE_MAKEFILE=FALSE \
   -D CMAKE_INSTALL_PREFIX:PATH=${TRILINOS_INSTALL} \
   -D CMAKE_BUILD_TYPE:STRING=${BUILD_TYPE} \
   -D BUILD_SHARED_LIBS=ON \
   -D CMAKE_USER_MAKE_RULES_OVERRIDE=${CMAKE_ROOT}/Cray-Intel.cmake \
   \
   -D CMAKE_C_COMPILER="cc" \
   -D CMAKE_CXX_COMPILER="CC" \
   -D CMAKE_Fortran_COMPILER="ftn" \
   -D CMAKE_C_FLAGS="$EXTRA_C_FLAGS" \
   -D CMAKE_CXX_FLAGS="$EXTRA_C_FLAGS" \
   -D CMAKE_Fortran_FLAGS="$EXTRA_F_FLAGS" \
   -D CMAKE_EXE_LINKER_FLAGS="$LINK_FLAGS" \
   \
   -D Trilinos_VERBOSE_CONFIGURE=OFF \
   -D Trilinos_ENABLE_ALL_PACKAGES=OFF \
   -D Trilinos_ENABLE_SECONDARY_TESTED_CODE=OFF \
   \
   -D Trilinos_ENABLE_TESTS=ON \
   -D Trilinos_ENABLE_EXAMPLES=OFF \
   -D DART_TESTING_TIMEOUT:STRING="600" \
   \
       -D Belos_ENABLE_TEUCHOS_TIME_MONITOR:BOOL=ON \
       -D Belos_Tpetra_Timers:BOOL=ON \
       -D Belos_ENABLE_TSQR=ON \
       -D Anasazi_ENABLE_TESTS=OFF \
       -D Anasazi_ENABLE_EXAMPLES=OFF \
   \
   -D Trilinos_ENABLE_EXPLICIT_INSTANTIATION=ON \
   -D Tpetra_INST_FLOAT=OFF \
   -D Tpetra_INST_DOUBLE=ON \
   -D Tpetra_INST_COMPLEX_FLOAT=OFF \
   -D Tpetra_INST_COMPLEX_DOUBLE=OFF \
   -D Tpetra_INST_INT_INT=ON \
   -D Tpetra_INST_INT_LONG=OFF \
   -D Tpetra_INST_INT_UNSIGNED=OFF \
   -D Tpetra_INST_INT_LONG_LONG=ON \
   -D Teuchos_ENABLE_LONG_LONG_INT=ON \
   -D Teuchos_ENABLE_COMPLEX=OFF \
   -D Zoltan_ENABLE_ULLONG_IDS=ON \
   \
   -D Trilinos_ENABLE_OpenMP=ON \
   -D TPL_ENABLE_Pthread=OFF \
   \
   -D Trilinos_ENABLE_Teuchos=ON \
   -D Trilinos_ENABLE_Epetra=ON \
   -D Trilinos_ENABLE_EpetraExt=ON \
   -D Trilinos_ENABLE_AztecOO=ON \
   -D Trilinos_ENABLE_Amesos=ON \
   -D Trilinos_ENABLE_Stratimikos=OFF \
   -D Trilinos_ENABLE_Anasazi=ON \
   -D Anasazi_ENABLE_RBGen=ON \
   -D Trilinos_ENABLE_Ifpack=ON \
   -D Trilinos_ENABLE_ML=ON \
   -D Trilinos_ENABLE_Teko=OFF \
   -D Trilinos_ENABLE_NOX=OFF \
   -D Trilinos_ENABLE_Thyra=OFF \
   -D Trilinos_ENABLE_Rythmos=OFF \
   -D Trilinos_ENABLE_Sacado=ON \
   -D Trilinos_ENABLE_Stokhos=OFF \
   -D Trilinos_ENABLE_Panzer=OFF \
   -D Trilinos_ENABLE_Tpetra=ON \
   -D Tpetra_INST_SERIAL=OFF \
   -D Tpetra_INST_OPENMP=ON \
   -D Trilinos_ENABLE_Belos=ON \
   -D Trilinos_ENABLE_Amesos2=ON \
   -D Amesos2_ENABLE_KLU2=ON \
   -D Trilinos_ENABLE_Ifpack2=ON \
   -D Trilinos_ENABLE_MueLu=OFF \
   -D Trilinos_ENABLE_Zoltan2=ON \
   -D Trilinos_ENABLE_STKMesh=ON \
   -D Trilinos_ENABLE_STKIO=ON \
   -D Trilinos_ENABLE_STKTransfer=ON \
   -D Trilinos_ENABLE_STKSearch=ON \
   -D Trilinos_ENABLE_STKUtil=ON \
   -D Trilinos_ENABLE_STKTopology=ON \
   \
   -D Trilinos_ENABLE_Kokkos=ON \
   -D Trilinos_ENABLE_KokkosCore=ON \
   -D Kokkos_ENABLE_Serial=OFF \
   -D Kokkos_ENABLE_OpenMP=ON \
   -D Kokkos_ENABLE_Pthread=OFF \
   -D Kokkos_ENABLE_Cuda=OFF \
   -D Kokkos_ENABLE_Cuda_UVM=OFF \
   \
   -D Trilinos_ENABLE_SEACAS=ON \
   -D TPL_ENABLE_X11=OFF \
   -D TPL_ENABLE_Matio=OFF \
   \
   -D Trilinos_ENABLE_Gtest=ON \
   \
   -D TPL_ENABLE_MPI=ON \
   -D MPI_USE_COMPILER_WRAPPERS=OFF \
   -D MPI_BASE_DIR:PATH=${MPI_DIR} \
   -D MPI_EXEC:PATH="${MPI_EXEC_CMD}" \
   -D MPI_EXEC_MAX_NUMPROCS:STRING="${MPI_EXEC_MAX_NUMPROCS}" \
   -D MPI_EXEC_NUMPROCS_FLAG:STRING="${MPI_EXEC_NUMPROCS_FLAG}" \
   \
   -D TPL_ENABLE_BLAS=ON \
   -D BLAS_LIBRARY_DIRS:PATH="${BLAS_DIR}/mkl/lib/intel64;${BLAS_DIR}/compiler/lib/intel64" \
   -D BLAS_LIBRARY_NAMES:STRING="mkl_intel_lp64;mkl_core;mkl_intel_thread;iomp5" \
   \
   -D TPL_ENABLE_LAPACK=ON \
   -D LAPACK_LIBRARY_DIRS:PATH="${LAPACK_DIR}/mkl/lib/intel64;${LAPACK_DIR}/compiler/lib/intel64" \
   -D LAPACK_LIBRARY_NAMES:STRING="mkl_intel_lp64;mkl_core;mkl_intel_thread;iomp5" \
   \
   -D TPL_ENABLE_Boost=ON \
   -D Boost_INCLUDE_DIRS:PATH=${BOOST_DIR}/include \
   \
   -D TPL_ENABLE_BoostLib=ON \
   -D BoostLib_INCLUDE_DIRS:PATH=${BOOST_DIR}/include \
   -D BoostLib_LIBRARY_DIRS:PATH=${BOOST_DIR}/lib \
   \
   -D TPL_ENABLE_Netcdf=ON \
   -D Netcdf_INCLUDE_DIRS:PATH="${NETCDF_DIR}/include;${HDF5_DIR}/include" \
   -D Netcdf_LIBRARY_DIRS:PATH="${NETCDF_DIR}/lib;${PNETCDF_DIR}/lib;${HDF5_DIR}/lib;${ZLIB_DIR}" \
   -D Netcdf_LIBRARY_NAMES:STRING="netcdf;pnetcdf;hdf5_hl;hdf5" \
   \
   -D TPL_ENABLE_METIS=OFF \
   -D METIS_INCLUDE_DIRS:PATH=${METIS_DIR}/include \
   -D METIS_LIBRARY_DIRS:PATH=${METIS_DIR}/lib \
   \
   -D TPL_ENABLE_ParMETIS=OFF \
   -D ParMETIS_INCLUDE_DIRS:PATH=${PARMETIS_DIR}/include \
   -D ParMETIS_LIBRARY_DIRS:PATH=${PARMETIS_DIR}/lib \
   \
   -D TPL_ENABLE_SuperLUDist=OFF \
   -D SuperLUDist_INCLUDE_DIRS:PATH=${SUPERLUDIST_DIR}/include \
   -D SuperLUDist_LIBRARY_DIRS:PATH=${SUPERLUDIST_DIR}/lib \
   -D SuperLUDist_LIBRARY_NAMES:STRING="superlu_dist_4.3" \
   \
   -D Trilinos_EXTRA_LINK_FLAGS:STRING="" \
   \
   ${EXTRA_ARGS} \
   ${TRILINOS_HOME}
```

@trilinos-zoltan2 @kddevin

**Comment by krcb** [comment]
@trillinos/zoltan2

**Comment by kddevin** [comment]
@trilinos/zoltan2

**Comment by kddevin** [comment]
Fixed by bd7dc13b6d2423d4e93e18fe9644870efd1a7dfd
